apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -464,27 +464,29 @@ private void EvictAndRefreshSessions()
 
                 void RemoveStaleOrExpiredItemsFromStack(ConcurrentStack<PooledSession> stack)
                 {
-                    int count = stack.Count;
-                    for (int i = 0; i < count; i++)
+                    if (!stack.IsEmpty)
                     {
-                        // This shouldn't happen, as nothing else should touch the stack while we're in a
-                        // lock, but we don't want to throw an exception
-                        if (!stack.TryPop(out var session))
+                        int count = stack.Count;
+                        PooledSession[] sessions = new PooledSession[count];
+                        if (stack.TryPopRange(sessions) > 0)
                         {
-                            break;
-                        }
-                        if (session.ShouldBeEvicted)
-                        {
-                            sessionsToEvict.AddLast(session);
-                        }
-                        else if (session.RequiresRefresh)
-                        {
-                            staleSessions.AddLast(session);
-                        }
-                        else
-                        {
-                            // The session is fine - put it back on the stack.
-                            stack.Push(session);
+                            for (int i = count - 1; i >= 0; i--)
```

Comment:
It's probably worth adding a comment here to explain what's going on, in terms of the order of elements in TryPopRange and what we're trying to achieve.
It's top-of-mind now, but it may not be in a year's time :)

Reply:
Yep, done